File: app/src/main/java/activitytest/example/com/roomdemo/main/viewModel/ConfigurationViewModel.java
package activitytest.example.com.roomdemo.main.viewModel;

import android.util.Log;

import activitytest.example.com.roomdemo.MyApplication;
import activitytest.example.com.roomdemo.main.MainRepository;

import activitytest.example.com.roomdemo.main.database.Configuration;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModel;

public class ConfigurationViewModel extends ViewModel {

    private static final String TAG  = ConfigurationViewModel.class.getName ();


    private LiveData<Configuration> configurationMutableLiveDataNet;
    private LiveData<Configuration> configurationMutableLiveDataBd;
    private LiveData<Integer> code;

    public void init(){
        Log.d ( TAG,"创建仓库" );
        MainRepository mainRepository = MainRepository.getInstance ();
        configurationMutableLiveDataNet = mainRepository.getConfiguration ();
        configurationMutableLiveDataBd = mainRepository.getConfigurationFromDB ();
        code = mainRepository.getCode ();
    }

    public LiveData<Configuration> getConfigurationNet(){
        Log.d ( TAG,"网络获取" );
        return configurationMutableLiveDataNet;
    }
    public LiveData<Configuration> getConfigurationDb(){
        Log.d ( TAG,"数据库获取" );

        return configurationMutableLiveDataBd;
    }

    public LiveData<Integer> getCode(){
        Log.d ( TAG,"获取状态码" );
        return code;
    }

}

File: app/src/main/java/activitytest/example/com/roomdemo/main/viewModel/ConfigurationViewModel.kt
package activitytest.example.com.roomdemo.main.viewModel

import activitytest.example.com.roomdemo.main.MainRepository
import activitytest.example.com.roomdemo.main.database.entity.Configuration
import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel

class ConfigurationViewModel : ViewModel() {
    private var configurationMutableLiveDataNet: LiveData<Configuration?>? = null
    private var configurationMutableLiveDataBd: LiveData<Configuration?>? = null
    var code: LiveData<Int?>? = null
    fun init() {
        Log.d(TAG, "创建仓库")
        val mainRepository: MainRepository? = MainRepository.instance
        if (mainRepository != null) {
            configurationMutableLiveDataNet = mainRepository.configuration
        }
        if (mainRepository != null) {
            configurationMutableLiveDataBd = mainRepository.configurationFromDB
        }
        if (mainRepository != null) {
            code = mainRepository.code
        }
    }

    val configurationNet: LiveData<Configuration?>?
        get() {
            Log.d(TAG, "网络获取")
            return configurationMutableLiveDataNet
        }
    val configurationDb: LiveData<Configuration?>?
        get() {
            Log.d(TAG, "数据库获取")
            return configurationMutableLiveDataBd
        }

    @JvmName("getCode1")
    fun getCode(): LiveData<Int?>? {
        Log.d(TAG, "获取状态码")
        return code
    }

    companion object {
        private val TAG = ConfigurationViewModel::class.java.name
    }
}